### DISCUSSION TITLE
x-show does not update element style as soon as data changes.

### DISCUSSION THREAD

**Discussion by Nick-LCY** [alpinejs/alpine#3738] [7. Bugs (archived)] [answered]
Alpine.js version: 3.5.0, CDN

In the following code, I defined three icons (represented by `span` for convenience): `(file|videos)`, `(file|subs)` and `(folder)`. I want to achieve the following effect: By clicking toggle button, the icon should switch between `(file|videos)` and `(folder)`. But in fact it will have four different states: `(file|videos)`, `(file|subs)(folder)` (yeah, two icons at the same time), `<empty>` (nothing inside) and `(folder)`.

```html
<html>
<header>
        <script defer src="https://cdn.jsdelivr.net/npm/alpinejs@3.5.0/dist/cdn.min.js"></script>
</header>
<body x-data="
    {
        videos: {type: 'folder', children: [
            {type: 'file', children: []},
        ]},
        subs: {type: 'folder', children: [
            {type: 'folder', children: []},
        ]},
        currentFolder: {},
        tab: 'videos',
    }">
    <div>
        <button x-data="{toggle: function() {
            if (this.tab == 'videos') {
                this.tab = 'subs';
                this.currentFolder = this.subs;
            } else {
                this.tab = 'videos';
                this.currentFolder = this.videos;
            }
        }}" @click="toggle()">
            <span>TOGGLE</span>
        </button>
        <span x-text="'Current tab: ' + tab"></span>
        <div>
            <template x-for="child in currentFolder.children">
                <button>
                    <span x-show="child.type == 'file' && tab == 'subs'">(file|subs)</span>
                    <span x-show="child.type == 'file' && tab == 'videos'">(file|videos)</span>
                    <span x-show="child.type == 'folder'">(folder)</span>
                </button>
            </template>
        </div>
    </div>
</body>
</html>
```

I'm not sure this is a bug or a feature (just like Vue cannot watch data changes inside an object). If this is a feature, is there any way to achieve the effect I want?

**Comment by SimoTod** [comment]
That version is 2 years old. Many bugs have been fixed since that time. I suggest you to try with a more recent version first.

**Reply by Nick-LCY** [reply]
Thanks for your advice, I have just test it under 3.13.0, but the problem still exists

**Reply by SimoTod** [reply, marked as answer]
Right, I think you just need to add a key to your x-for: https://codepen.io/SimoTod/pen/KKbzxBY
I've used child/type in this simplified example but I assume you will have something better in the real app such as a unique ID or file/folder name
